pytorch_lightning/trainer/deprecated_api.py

```diff
@@ -23,7 +23,7 @@ class DeprecatedDistDeviceAttributes:
 
     @property
     def on_cpu(self) -> bool:
-        # rank_zero_warn("Internal: `on_cpu` is deprecated in v1.1 and will be removed in v1.2.", DeprecationWarning)
+        rank_zero_warn("Internal: `on_cpu` is deprecated in v1.2 and will be removed in v1.4.", DeprecationWarning)
```

Comment:
What will be the alternative when removed? (for users)

`self.trainer._device_type == DeviceType.CPU`?

If that's the case, we should remove the underscore...

Reply:
Why do we want to remove this at all?

Reply:
good point, shall we work with the Enum directly or keep some properties?
cc: @PyTorchLightning/core-contributors

Reply:
I like the Enum. The enum could also have functions for utility checks, for example:

```python

class DistributedType(LightningEnum):
    DP = 'dp'
    ...

    def is_distributed(self):
        return self in (DistributedType.DDP, DistributedType.DDP_SPAWN, DistributedType.DDP2)
```

so users could do:

`if self.trainer.device_type.is_distributed()`

Reply:
that is a nice feature, lets make it as follow-up pr :]

Reply:
We should make sure we don't use these properties internally otherwise users gets spammed with deprecation warnings and they can't do anything about it.

This already happens in some places with other deprecation warnings and this is very confusing to the user (warning utils)

Reply:
I passed all actual `release/1.2-dev` codebase and it was pruned...